File: drills/ts/harnesses/nc48_valid_parenthesis_string.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.checkValidString !== "function") throw new Error("__signature_mismatch__");
  return [
    ["checkValidString('()')", show(true), show(solution.checkValidString('()'))],
    ["checkValidString('(*)')", show(true), show(solution.checkValidString('(*)'))],
    ["checkValidString('(*))')", show(true), show(solution.checkValidString('(*))'))],
    ["checkValidString(')(')", show(false), show(solution.checkValidString(')('))],
    ["checkValidString('')", show(true), show(solution.checkValidString(''))],
    ["checkValidString('*')", show(true), show(solution.checkValidString('*'))],
    ["checkValidString(')*')", show(false), show(solution.checkValidString(')*'))],
    ["checkValidString('(*()')", show(true), show(solution.checkValidString('(*()'))],
  ];
}

File: drills/go/harnesses/nc48_valid_parenthesis_string.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("checkValidString('()')", true, checkValidString("()")),
			tc("checkValidString('(*)')", true, checkValidString("(*)")),
			tc("checkValidString('(*))')", true, checkValidString("(*))")),
			tc("checkValidString(')(')", false, checkValidString(")(")),
			tc("checkValidString('(((**')", false, checkValidString("(((**")),
			tc("checkValidString('**((')", false, checkValidString("**((")),
			tc("checkValidString('')", true, checkValidString("")),
		}
	})
}
